db/blob/db_blob_compaction_test.cc

```diff
@@ -390,6 +390,34 @@ TEST_F(DBBlobCompactionTest, CompactionFilterReadBlobAndKeep) {
   Close();
 }
 
+TEST_F(DBBlobCompactionTest, MergeBlobWithBase) {
+  Options options = GetDefaultOptions();
+  options.create_if_missing = true;
+  options.enable_blob_files = true;
+  options.min_blob_size = 0;
+  options.merge_operator = MergeOperators::CreateStringAppendOperator();
+  options.disable_auto_compactions = true;
+
+  DestroyAndReopen(options);
+
+  ASSERT_OK(Put("Key1", "v1"));
+  ASSERT_OK(Flush());
+  MoveFilesToLevel(2);
```

Comment:
Are these `MoveFilesToLevel` calls necessary? I think we could simply create a bunch of L0 files and then compact them together with a single `CompactRange` call (like the one you have below)...

Reply:
I tried initially `Flush()` after each `put/merge` and then `CompactRange` in the end but then it doesn't call the required code that we want to test in `merge_helper.cc`. I didn't look in depth why so. May be I am missing some option to be set in the test. I will remove the last call  `CompactRange` as it doesn't call the required code (MoveFilesToLevel calls the code)

Reply:
Just taking a stab in the dark here but the issue you describe might be caused by trivial moves. You could try using two keys instead of one (i.e. putting the same two keys in each file).

Reply:
P.S. You could try running the test with `KEEP_DB=1` to confirm this.

Reply:
Ah ok. I will try this  with `KEEP_DB=1`. Thanks!

Reply:
Yes, It was because of trivial move.